### ISSUE TITLE
No run button (again, but different)

### ISSUE THREAD

**Issue by JPvRiel** [tillahoffmann/obsidian-jupyter#50]
**Describe the bug**

Run button is not added.

Context:

- Seems that the plugin does not properly load in/get the the run button added to the editors DOM, despite the `test.md` only containing a python code block.
- Ubuntu 20.04
- Tested both 0.14.15 and latest 0.15.9 Obsidian AppImage
- Tested with a clean vault, only obsidian-jupyter installed.
-  Tested both system `python3` (3.8.10) and alternate upgraded python 3.10.4. Issue persists.

Cannot find any info in developer console output as to why "Run" button is not added.

**Screenshots**

![image](https://user-images.githubusercontent.com/9722317/187186349-5698703f-af65-4237-8c09-5b10a0497479.png)

**Console output**

After the screenshot, I made sure I upgraded Obsidian to the latest AppImage and ran the test again to produce the console output.

Test output works fine?

```console
loading jupyter plugin
plugin:obsidian-jupyter:3247 python script exists at /home/a211278l/Scratch/Obsidian/Test/.obsidian/plugins/obsidian-jupyter/obsidian-jupyter.py
app.js:1 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
n @ app.js:1
e @ app.js:1
e @ app.js:1
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
app.js:1 [Violation] Added non-passive event listener to a scroll-blocking 'touchmove' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
n @ app.js:1
e @ app.js:1
e @ app.js:1
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
app.js:1 [Violation] 'click' handler took 273ms
[Violation] Forced reflow while executing JavaScript took 213ms
plugin:obsidian-jupyter:3220 created new client for doc test-document using interpreter /usr/local/bin/python3.10
plugin:obsidian-jupyter:3146 INFO:obsidian-jupyter:started server for document test-document

plugin:obsidian-jupyter:3146 INFO:obsidian-jupyter:received request: {"id":"bf434af0-115c-4f56-936a-4bbfebf9a4ec","body":{"command":"execute","source":"1 + 1"}}


plugin:obsidian-jupyter:3146 INFO:traitlets:Executing notebook with kernel: 

plugin:obsidian-jupyter:3146 INFO:obsidian-jupyter:sent response: {"id": "bf434af0-115c-4f56-936a-4bbfebf9a4ec", "body": "<div class=\"cell border-box-sizing code_cell rendered\">\n<div class=\"input\">\n<div class=\"prompt input_prompt\">In&nbsp;[1]:</div>\n<div class=\"inner_cell\">\n    <div class=\"input_area\">\n<div class=\" highlight hl-ipython3\"><pre><span></span><span class=\"mi\">1</span> <span class=\"o\">+</span> <span class=\"mi\">1</span>\n</pre></div>\n\n    </div>\n</div>\n</div>\n\n<div class=\"output_wrapper\">\n<div class=\"output\">\n\n\n<div class=\"output_area\">\n\n    <div class=\"prompt output_prompt\">Out[1]:</div>\n\n\n\n\n<div class=\"output_text output_subarea output_execute_result\">\n<pre>2</pre>\n</div>\n\n</div>\n\n</div>\n</div>\n\n</div>\n\n\n"}

plugin:obsidian-jupyter:3135 received response {id: 'bf434af0-115c-4f56-936a-4bbfebf9a4ec', body: '<div class="cell border-box-sizing code_cell rende…>2</pre>\n</div>\n\n</div>\n\n</div>\n</div>\n\n</div>\n\n\n'}
plugin:obsidian-jupyter:3339 Received response <div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="mi">1</span> <span class="o">+</span> <span class="mi">1</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt output_prompt">Out[1]:</div>




<div class="output_text output_subarea output_execute_result">
<pre>2</pre>
</div>

</div>

</div>
</div>

</div>



plugin:obsidian-jupyter:3146 INFO:obsidian-jupyter:shutting down kernel...

plugin:obsidian-jupyter:3146 INFO:obsidian-jupyter:exiting...
```

I tried using options to set the interpreter in the note front-matter.

```
obsidian-jupyter:
 interpreter: /usr/bin/python3
---

And I tried opening and closing the note. The only output I got on the console, probably unrelated, was:

```
app.js:1 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
app.js:1 [Violation] Added non-passive event listener to a scroll-blocking 'touchmove' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
```

**Python dependencies**

```console
/usr/local/bin/python3.10 -m pip freeze
adal==1.2.7
ansible-core==2.12.7
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
asn1crypto==1.5.1
asttokens==2.0.8
attrs==22.1.0
azure-common==1.1.28
azure-core==1.23.0
azure-identity==1.8.0
backcall==0.2.0
beautifulsoup4==4.11.1
bleach==5.0.1
blessings==1.7
boto3==1.21.13
botocore==1.24.13
bpython==0.22.1
certifi==2021.10.8
certvalidator==0.11.1
cffi==1.15.1
chardet==5.0.0
charset-normalizer==2.0.12
click==8.0.4
colorama==0.4.5
colorful==0.5.4
cryptography==36.0.1
curtsies==0.3.10
cwcwidth==0.1.6
cycler==0.11.0
debugpy==1.6.3
decorator==5.1.1
defusedxml==0.7.1
entrypoints==0.4
et-xmlfile==1.1.0
executing==0.10.0
fastjsonschema==2.16.1
Flask==2.0.3
fonttools==4.29.1
future==0.18.2
greenlet==1.1.2
idna==3.3
install==1.3.5
ipykernel==6.15.1
ipython==8.4.0
ipython-genutils==0.2.0
ipywidgets==8.0.1
isodate==0.6.1
itsdangerous==2.1.1
jarowinkler==1.0.2
jedi==0.18.1
Jinja2==3.1.2
jmespath==1.0.1
joblib==1.1.0
jsonschema==4.14.0
jupyter==1.0.0
jupyter-console==6.4.4
jupyter-core==4.11.1
jupyter_client==7.3.5
jupyterlab-pygments==0.2.2
jupyterlab-widgets==3.0.2
kiwisolver==1.3.2
Kqlmagic==0.1.114.post20
lxml==4.9.1
Markdown==3.3.6
MarkupSafe==2.1.1
matplotlib==3.5.1
matplotlib-inline==0.1.6
mistune==2.0.4
msal==1.17.0
msal-extensions==0.3.1
msrest==0.6.21
msrestazure==0.6.4
nbclient==0.6.7
nbconvert==7.0.0
nbformat==5.4.0
nest-asyncio==1.5.5
notebook==6.4.12
numpy==1.22.2
oauthlib==3.2.0
openpyxl==3.0.10
oscrypto==1.3.0
packaging==21.3
pandas==1.4.2
pandocfilters==1.5.0
parso==0.8.3
passlib==1.7.4
password-strength==0.0.3.post2
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.0.1
plotly==5.6.0
polyfuzz==0.3.4
portalocker==2.4.0
prettyprinter==0.18.0
prettytable==3.2.0
prometheus-client==0.14.1
prompt-toolkit==3.0.30
psutil==5.9.1
ptyprocess==0.7.0
pure-eval==0.2.2
pycparser==2.21
Pygments==2.13.0
PyJWT==2.3.0
PyNaCl==1.5.0
pyparsing==3.0.9
pyperclip==1.8.2
pyrsistent==0.18.1
python-dateutil==2.8.2
python-magic==0.4.27
python-whois==0.7.3
pytz==2021.3
pytz-deprecation-shim==0.1.0.post0
pyxdg==0.27
PyYAML==6.0
pyzmq==23.2.1
qtconsole==5.3.1
QtPy==2.2.0
rapidfuzz==2.0.7
requests==2.27.1
requests-oauthlib==1.3.1
resolvelib==0.5.4
ruamel.yaml==0.17.20
ruamel.yaml.clib==0.2.6
s3transfer==0.5.2
scikit-learn==1.0.2
scipy==1.8.0
seaborn==0.11.2
Send2Trash==1.8.0
shtab==1.5.5
six==1.16.0
soupsieve==2.3.2.post1
stack-data==0.4.0
tenacity==8.0.1
termcolor==1.1.0
terminado==0.15.0
testpath==0.6.0
threadpoolctl==3.1.0
tinycss2==1.1.1
tldr==3.1.0
tornado==6.2
tqdm==4.64.0
traitlets==5.3.0
typing_extensions==4.1.1
tzdata==2022.1
tzlocal==4.2
urllib3==1.26.8
vcert==0.15.0
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==2.0.3
widgetsnbextension==4.0.2
```

**Related**

These issues are similar. However, in my testing, I tested without any plugins installed, so there should be no conflict.

- <https://github.com/tillahoffmann/obsidian-jupyter/issues/4>
- <https://github.com/tillahoffmann/obsidian-jupyter/issues/25>
